modules/programs/waybar.nix

```diff
@@ -0,0 +1,388 @@
+        example = "waybar-1";
+      };
+
+      gtk-layer-shell = mkOption {
+        type = nullOr bool;
+        default = null;
+        example = false;
+        description =
+          "Option to disable the use of gtk-layer-shell for popups.";
+      };
+    };
+  };
+in {
+  meta.maintainers = [ hm.maintainers.berbiche ];
+
+  options.programs.waybar = {
+    enable = mkEnableOption "Waybar";
+
+    package = mkOption {
+      type = nullOr package;
+      default = pkgs.waybar;
+      defaultText = literalExample "${pkgs.waybar}";
+      description = ''
+        Waybar package to use. Set to <code>null</code> to use the default module.
+      '';
+    };
+
+    settings = mkOption {
+      # maybe-todo: improve type when https://github.com/NixOS/nixpkgs/pull/75584 is merged
+      type = listOf waybarBarConfig;
+      default = [ ];
+      description = ''
+        Configuration for Waybar, see <link xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        [
+          {
+            layer = "top";
+            position = "top";
+            height = 30;
+            # Specify outputs to restrict to certain outputs, otherwise show on all outputs
+            output = [
+              "eDP-1"
+              "DP-1"
+            ];
+            modules-left = [ "sway/workspaces" "sway/mode" "wlr/taskbar" ];
+            modules-center = [ "sway/window" "custom/hello-from-waybar" ];
+            modules-right = [ "mpd" "custom/mymodule#with-css-id" "temperature" ];
+            modules = {
+              "sway/workspaces" = {
+                disable-scroll = true;
+                all-outputs = true;
+              };
+              "custom/hello-from-waybar" = {
+                format = "hello {}";
+                max-length = 40;
+                interval = "once";
+                # You may be interested in using symlinkJoin to merge all scripts in one derivation
+                # to have them all under one directory structure in the nix store
+                exec = "''${pkgs.writers.writeBashBin "hello-from-waybar" '''
+                  echo "from within waybar"
+                '''}/bin/hello-from-waybar";
+              };
+            };
+          }
+        ]
+      '';
+    };
+
+    systemd = {
+      enable = mkEnableOption "Waybar Systemd integration";
+
+      withSwayIntegration = mkOption {
+        type = bool;
+        default = false;
+        example = true;
+        description =
+          "Whether to bind the systemd service to the sway-session.target instead of graphical-session.target.";
+      };
+    };
+
+    style = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        CSS style of the bar.
+        See <link xlink:href="https://github.com/Alexays/Waybar/wiki/Configuration/>"/> for the documentation.
```

Comment:
I think the URL should be https://github.com/Alexays/Waybar/wiki/Configuration? The `/>` part might be a mistake.

Reply:
Thanks fixed